presto-hive/src/main/java/com/facebook/presto/hive/metastore/SemiTransactionalHiveMetastore.java

```diff
@@ -1395,7 +1399,7 @@ private void executeIrreversibleMetastoreOperations()
     }
 
     @GuardedBy("this")
-    private void rollbackShared()
+    private void deleteStagingData()
```

Comment:
For other mode, such as `DIRECT_TO_TARGET_EXISTING_DIRECTORY`,  it's not deleting staging data , but to delete new created files in this directory. So I would keep the `rollbackShared` name.

Reply:
Technically it is still deleting staging data. The only difference is that with first two option - the entire directory is deleted, when in case of the `DIRECT_TO_TARGET_EXISTING_DIRECTORY` staging data is deleted file by file. So, i would recommend to keep the `deleteStagingData` name, as `rollbackShared` may imply that some other actions (for example removing some metastore records) are done within.

Reply:
It's true `deleteStagedData` more accurately describes what this method is doing, but it becomes confusing because there's a write mode that talks about staged data where it means something slightly different.  I'm going to leave this as 'rollbackShared' since I refactored to move the property check inside the method.